presidio-analyzer/presidio_analyzer/input_validation/yaml_recognizer_models.py

```diff
@@ -0,0 +1,487 @@
+"""Pydantic models for YAML recognizer configurations."""
```

Comment:
This file contains the pydantic equivalent of the yaml structures. As these are not 1:1 with Presidio objects in some cases, there's an intermediate pydantic representation. The existing layer (AnalyzerEngineProvider, NlpEngineProvider, RecognizerRegistryProvider) still handles the actual creation of objects.